services/frontend-v3/src/components/searchFilter/SearchFilter.jsx

```diff
@@ -14,11 +15,41 @@ const SearchFilter = ({ history, changeLanguage }) => {
   const [branchOptions, setBranchOptions] = useState([]);
   const [locationOptions, setLocationOptions] = useState([]);
   const [classOptions, setClassOptions] = useState([]);
+  const [previousValues, setPreviousValues] = useState(null);
 
   const toggle = () => {
     setExpand(!expand);
   };
 
+  const getPreviousValues = useCallback(() => {
```

Comment:
I would rename "getPreviousValues" to something like "getSearchFeildValues" to be more descriptive and maybe add a small comment about what the funstion does